### PR TITLE
docs(style): Bold section names

### LINKED ISSUE #5614: docs(style): Bold section names
It'd be nice to bold the sections, e.g. "Introduction" and "Getting Started" in

Screenshot 2024-07-30 at 12 24 20 PM

### PR DESCRIPTION
**Pull request by eth3lbert**
## Summary

Closes #5614 .

## Test Plan

Run `mkdocs serve -f mkdocs.public.yml` to see the changes.

### CODE REVIEW

**Comment by eth3lbert** [comment]
Some screenshots:

image
image

**Comment by eth3lbert** [comment]
> Should we make the color match the other bolded headings?

Hmm, this will make the active one no different from others. Maybe it's not a good idea. :sweat_smile:

**Comment by zanieb** [comment]
Presumably we could use `.md-nav__link--active` to retain the active color change, right?

Alternatively we could explore a less heavy bold font-weight but the default family (at least on my platform) doesn't support it.

**Comment by eth3lbert** [comment]
Also, if you think it's too bold, you could change the value to something(500, 600, ...) that larger than 400 (normal). Currently, it's 700 (bold).

**Comment by zanieb** [comment]
There's also 

```diff
diff --git a/mkdocs.template.yml b/mkdocs.template.yml
index 26d01d0d..14c01bcf 100644
--- a/mkdocs.template.yml
+++ b/mkdocs.template.yml
@@ -8,6 +8,7 @@ theme:
     - navigation.instant.prefetch
     - navigation.instant.progress
     - navigation.expand
+    - navigation.sections
     - navigation.indexes
     - navigation.tracking
     - content.code.annotate
```

I'm not sure if that's better or not?

**Comment by eth3lbert** [comment]
> ```diff
> - navigation.sections
> ```

image

This lack of indentation.

And style preference that varies from person to person. I suppose we could let others choose.

**Comment by zanieb** [comment]
Yeah I'm not sure what's best. @charliermarsh and @dhruvmanila curious for your thoughts.

**Comment by dhruvmanila** [comment]
I think `navigation.sections` looks good as we're already using `navigation.expand`. So, `navigation.sections` would provide both the bold section names _and_ expanded navigation. It does have a limitation that one cannot close any specific section but I doubt that readers would close all sections except for the one they're focused in. I do like the indentation but couldn't find any reference for it in the mkdocs-material documentation.

tldr, my opinion is to replace `navigation.expand` with `navigation.sections`.

**Comment by eth3lbert** [comment]
> I do like the indentation but couldn't find any reference for it in the mkdocs-material documentation.

I think this can be achieved by the css.

**Comment by zanieb** [comment]
Superseded by #5674

**Comment by zanieb** [comment]
Thanks for exploring this though
